velox/exec/MergingVectorOutput.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::exec {
+
+class MergingVectorOutput {
+ public:
+  MergingVectorOutput(
+      velox::memory::MemoryPool* pool,
+      int64_t preferredOutputBatchBytes,
+      int32_t preferredOutputRows,
+      int32_t minOutputBatchRows);
+
+  void addVector(RowVectorPtr input);
+
+  RowVectorPtr getOutput();
```

Comment:
I'm wondering if the API can be simplified to just addVector(input) and getOutput(bool noMoreInput) methods. 

getOutput method would take a boolean indicating whether more data is expected to be added via addVector and return null if there is no more data to produce or we haven't accumulated enough just yet.

Reply:
done